[sc-21418] adds retry on network failures

dlively1 opened the pull request:
Adds retry capabilities for network failures

dlively1 committed:
[sc-21418] adds retry on network failures

namoscato commented on package-lock.json:
It looks like this project is using Yarn, so we won't want to commit this.

dlively1 replied:
🤦 I thought that was already there and was wondering what was going on. Let me patch this up

namoscato commented on yarn.lock:
I think this is using Yarn v1 / lockfile v1. Is that what you were using to install the package?

I'm getting a number of changes when I run through these steps:

1. install Yarn v1:

    ```
    npm install --global yarn
    ```

1. add package:

    ```
    yarn add @adobe/node-fetch-retry
    ```

dlively1 committed:
[sc-21418] remove package-lock.json

dlively1 commented on node_modules/.yarn-integrity:
```diff
@@ -1,19 +1,13 @@
 {
-  "systemParams": "darwin-x64-108",
+  "systemParams": "darwin-arm64-108",
```
Any insight into this change @namoscato ? I'm actually not super familiar with yarn.

namoscato replied:
I'm not intimately familiar with it either but don't think we have control over it given we're running this on M1s. It looks like this changed recently to x64 in https://github.com/UnlyEd/github-action-await-vercel/pull/89, so let's roll with this proposal.

codybswaney replied:
Curious.... how do we mean "we're running this on M1s"? Won't this be invoked via x86_64 architectures in github actions? 

Admittedly, I'm not familiar with yarn so I may be misunderstanding what this represents.

namoscato replied:
Oh sorry, to be clear – we ran the `yarn add` command on an M1 to install this new package locally, which resulted in this `.yarn-integrity` change, which is committed to version control.

namoscato committed:
Minimize yarn.lock changes

Reinstall new package on:
$ yarn --version
1.22.19

$ rm -rf node_modules/
$ git checkout main -- yarn.lock
$ yarn add @adobe/node-fetch-retry

namoscato committed:
Revert main.test.ts change

namoscato commented on yarn.lock:
d2f34783791d3a12854d0cc534bc3253226167a8 reinstalls the new package from a clean slate to minimize these `yarn.lock` changes (and the commit description enumerates the commands I ran).

dlively1 replied:
I'm confused as thats exactly what I ran. I even have the same yarn version. 😕 

Anyway thanks for the review and ✋
